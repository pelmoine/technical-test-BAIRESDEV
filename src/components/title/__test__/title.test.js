import React from 'react';
import ReactDom from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Title from '../title-component';

afterEach(cleanup);

it("render whitout crashing", () => {
    const div = document.createElement('div');
    ReactDom.render(<Title title='' />, div);
});

it("renders title correctly", () => {
    const {getByTestId} = render(<Title title='test' />);
    expect(getByTestId('title')).toHaveTextContent('test');
});

it("matches title snapshot", () => {
    const tree = renderer.create(<Title title='test' />).toJSON();
    expect(tree).toMatchSnapshot();
});