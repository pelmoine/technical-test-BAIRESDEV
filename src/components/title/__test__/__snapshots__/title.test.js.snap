// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches title snapshot 1`] = `
<div>
  <span
    data-testid="title"
  >
    test
  </span>
</div>
`;
